### ISSUE TITLE
Concatenated string goes in one word lines

### ISSUE THREAD

**Issue by hyperknot** [prettier/prettier#3913]
**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA3AhgJwAQEcCucmAntgLzYDkA9JdgNTYC2AllAB4NUA0djrUUo0q8uACibp2nALTM27AJRcRfbBKmk5A4ooA6UPTGHUADmBhI1ARgAMt7NWwBnMOgA2cFdVu0uBDxYYISoAOjU4dngoZxZoEG4QCFMYOJjkUCxMCAB3AAUsBGdkEHRUCBYAEwTS5xhkADMPZzhEgCNMdDAAazgYAGVTLrYAc2QYTEJEyogwRubWkDYWzBg8zpHJefcWxIArZ3YAIU6evv70JjgAGTY4bd2k-BhTZ4AmB8WhzBWStvQ2sR3PFEqZMGwYAB1KowAAWyGsb1soOyLUhnVMJTBcBWqHuiUwcAILEJ63Qm3Qn0SLVY40mi1iUBGngAivgIPAqSAYADoZU4cg3okJugWO5RgBhCBMLYoKDQfEgfAtAAqAOKSCaOzgAF8dUA)
```sh
--print-width 120
```

**Input:**
```jsx
var query = '/' + minx + ',' + miny + ',' + (maxx - minx) + ',' + (maxy - miny)
	+ '/pct:' + 100 / scale + '/0/' + quality + '.' + extension
```

**Output:**
```jsx
var query =
  "/" +
  minx +
  "," +
  miny +
  "," +
  (maxx - minx) +
  "," +
  (maxy - miny) +
  "/pct:" +
  100 / scale +
  "/0/" +
  quality +
  "." +
  extension;

```

**Expected behavior:**
This is quite not pretty like this. Something in 2 or maybe 3 would be reasonable, but not this.

**Comment by josephfrazier** [comment]
For what's it worth, writing this as a template literal makes it wrap only as much as needed to fit the `--print-width`:

**Input**:

```js
const query = `/${minx},${miny},${maxx - minx},${maxy - miny}/pct:${100 / scale}/0/${quality}.${extension}`;
```

**Output**:

```js
const query = `/${minx},${miny},${maxx - minx},${maxy - miny}/pct:${100 /
  scale}/0/${quality}.${extension}`;
```

[playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBHArnAJwE9MBeTAAwHoASYAWwEsoAPAXwBo6mojPuAhixaYAtJh7suDISXE8+VAA5gYSOgEYADFsxVMaMAIA2cNlS21geE4xh8AdHTgt4UNI2hsKAbhAcQCCUYT3dkUCUCZhgAdUYAExgAC2QADi0AmAEAIzjElKQAJgCPKABzUwBFXAh4ZAAzEzQ4TIIBRmNmMoBhCHp6AWQQKGg4fxBstrAAazgYAGUlATAu5BgCfACAKzQWACEp2YWBejgAGWYxpEbjZoClgmaCIeycomNoceamNY2WkFwzQAKjk0A0mv9mE8YAAFNplAbg27-AhwPCMVFwgQIwbXCH3AgQZoxNpKIaROBPABuYwCAgIhIA7jD6QgwSgBFSIAlxvEIGAkXcQAIMIL-hBcDAlJLCmK2GwgA) (note that `--print-width 120` wouldn't wrap in this case)

**Comment by hyperknot** [comment]
I totally agree that I wouldn't write this string like this, but this is from a library which I have to use verbatim as rewriting it could introduce bugs. Also, template literal support is not universal yet.

**Comment by josephfrazier** [comment]
Of course, I didn't mean to imply that you should have to rewrite your code. Instead, maybe Pretter could use similar logic for wrapping binary operators.

**Comment by vjeux** [comment]
You can workaround by adding parenthesis fwiw:

**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA3AhgJwAQEcCucmAntgLzYDkA9JdgNTYAUAtgJZQAeDVANHY3ZRSjSvx6t0nbgFpsQzgEpFEsQOYsppOUOKKAOlH0xR1AA5gYSdQEYADHezVsAZzDoANnB407tHgSebDAiVAB06nCc8FAubNCKILwgEGYw8bHIoFiYEADuAApYCC7IIOioEGwAJknlLjDIAGaeLnDJAEaY6GAA1nAwAMpmPRwA5sgwmITJ1RBgza3tIBxtmDAF3WOaix5tyQBWLpwAQt19A4PoLHAAMhxwu-sp+DBmrwBMT8sjmGtlHXQHWIHmgdTMmA4MAA6jUYAALZA2D52ZIQiBtaHdMxlCFwNaoR7JTBwAhsEmbdDbdDfZJtdiTabLOJQMZeACK+Ag8FpIBgQNh1QRyA+ySm6DYHnGAGEICwdigoNAiSB8G0ACpA0pIFp7OAAX31QA)
```sh
--print-width 120
```

**Input:**
```jsx
var query = '/' + (minx + ',' + miny + ',' + (maxx - minx)) + (',' + (maxy - miny)
	+ '/pct:' + 100 / scale + '/0/' + quality + '.' + extension)
```

**Output:**
```jsx
var query =
  "/" +
  (minx + "," + miny + "," + (maxx - minx)) +
  ("," + (maxy - miny) + "/pct:" + 100 / scale + "/0/" + quality + "." + extension);

```
